src/commands/Miscellaneous/discordstatus.ts

```diff
@@ -15,27 +15,121 @@ export default class extends SteveCommand {
 
 	public async run(msg: KlasaMessage): Promise<Message> {
 		const url = 'https://srhpyqt94yxb.statuspage.io/api/v2/summary.json';
-		const embedData = msg.language.tget('commandDiscordStautsEmbed');
-
-		return fetch(url, { method: 'Get' })
-			.then(res => res.json())
-			.then(json => {
-				const embed = new MessageEmbed()
-					.setTitle(json.status.description)
-					.setDescription(embedData.decription(json.status.indicator))
-					.addFields(json.components.map((component: { name: any; status: any }) => ({
+		const embedData = msg.language.tget('commandDiscordStatusEmbed');
+
+		const currentStatus = await fetch(url, { method: 'Get' }).then(r => r.json()) as DiscordStatus;
+
+		if (!currentStatus) {
+			return msg.channel.send(msg.language.tget('commandDiscordStatusError'));
+		}
+
+		const fields: Array<EmbedField> = [];
+
+		if (currentStatus.components?.every(component => component.status === 'operational')) {
+			fields.push({
+				name: embedData.fields.operational.title,
+				value: embedData.fields.operational.value,
+				inline: false
+			});
+		} else {
+			currentStatus.components.forEach(component => {
+				if (component.status !== 'operational') {
+					fields.push({
 						name: component.name,
 						value: component.status,
 						inline: true
-					})))
-					.setFooter(embedData.footer(formatDate(json.page.updated_at)));
-
-				return msg.channel.send(embed);
-			})
-			.catch(err => {
-				console.log(err);
-				return msg.sendLocale('commandDiscordStatusError');
+					});
+				}
 			});
+		}
+
+		if (currentStatus.scheduled_maintenances.length > 0) {
+			fields.push({
+				name: embedData.fields.maintenance.title,
+				value: currentStatus.scheduled_maintenances.map(m => embedData.fields.maintenance.value(m.name, m.impact)).join('\n'),
+				inline: false
+			});
+		}
+
+		const embed = new MessageEmbed()
+			.setTitle(currentStatus.status.description)
+			.setDescription(embedData.description(currentStatus.incidents[0]
+				? currentStatus.incidents.map(i => i.name).join('\n')
+				: embedData.noIncidents))
+			.addFields(fields)
+			.setThumbnail('https://discord.com/assets/2c21aeda16de354ba5334551a883b481.png')
+			.setTimestamp()
+			.setFooter(embedData.footer(formatDate(new Date(currentStatus.page.updated_at))))
+			.setColor('BLURPLE');
+
+		return msg.channel.send(embed);
+
 	}
 
 }
+
+interface DiscordStatus {
+	page: Page;
+	status: Status;
+	components: (ComponentsEntity)[];
```

Comment:
shouldn't need the parens here (and in the rest of the interfaces)

Reply:
Resolved by [029eebb](https://github.com/tuataria/steve/pull/63/commits/029eebb0e89e4770e99b06fc6556b9367f350771)